featuretools/tests/entityset_tests/test_es.py

```diff
@@ -878,98 +879,223 @@ def test_already_sorted_parameter():
-
-#     assert 10 not in es_1['log'].last_time_index.index
-#     assert 10 in es_2['log'].last_time_index.index
-#     assert 9 in es_1['log'].last_time_index.index
-#     assert 9 not in es_2['log'].last_time_index.index
-#     assert not es.__eq__(es_1, deep=True)
-#     assert not es.__eq__(es_2, deep=True)
-
-#     # make sure internal indexes work before concat
-#     regions = es_1.query_by_values('customers', ['United States'], variable_id=u'région_id')
-#     assert regions.index.isin(es_1['customers'].df.index).all()
-
-#     assert es_1.__eq__(es_2)
-#     assert not es_1.__eq__(es_2, deep=True)
-
-#     old_es_1 = copy.deepcopy(es_1)
-#     old_es_2 = copy.deepcopy(es_2)
-#     es_3 = es_1.concat(es_2)
-
-#     assert old_es_1.__eq__(es_1, deep=True)
-#     assert old_es_2.__eq__(es_2, deep=True)
-
-#     assert es_3.__eq__(es)
-#     for entity in es.entities:
-#         df = es[entity.id].df.sort_index()
-#         df_3 = es_3[entity.id].df.sort_index()
-#         for column in df:
-#             for x, y in zip(df[column], df_3[column]):
-#                 assert ((pd.isnull(x) and pd.isnull(y)) or (x == y))
-#         orig_lti = es[entity.id].last_time_index.sort_index()
-#         new_lti = es_3[entity.id].last_time_index.sort_index()
-#         for x, y in zip(orig_lti, new_lti):
-#             assert ((pd.isnull(x) and pd.isnull(y)) or (x == y))
-
-#     es_1['stores'].last_time_index = None
-#     es_1['test_entity'].last_time_index = None
-#     es_2['test_entity'].last_time_index = None
-#     es_4 = es_1.concat(es_2)
-#     assert not es_4.__eq__(es, deep=True)
-#     for entity in es.entities:
-#         df = es[entity.id].df.sort_index()
-#         df_4 = es_4[entity.id].df.sort_index()
-#         for column in df:
-#             for x, y in zip(df[column], df_4[column]):
-#                 assert ((pd.isnull(x) and pd.isnull(y)) or (x == y))
-
-#         if entity.id != 'test_entity':
-#             orig_lti = es[entity.id].last_time_index.sort_index()
-#             new_lti = es_4[entity.id].last_time_index.sort_index()
-#             for x, y in zip(orig_lti, new_lti):
-#                 assert ((pd.isnull(x) and pd.isnull(y)) or (x == y))
-#         else:
-#             assert es_4[entity.id].last_time_index is None
+def test_concat_simple(es):
+    if ks and any(isinstance(df, ks.DataFrame) for df in es.dataframes):
+        pytest.xfail("Koalas deepcopy fails")
+
+    first_es = copy.deepcopy(es)
+    for df in first_es.dataframes:
+        new_df = df.loc[[], :]
+        first_es.update_dataframe(df.ww.name, new_df)
+
+    second_es = copy.deepcopy(es)
+
+    # set the data description
+    first_es.metadata
+
+    new_es = first_es.concat(second_es)
+
+    assert new_es == es
+    assert new_es._data_description is None
+    assert first_es._data_description is not None
+
+
+def test_concat_inplace(es):
+    first_es = copy.copy(es)
+    second_es = copy.copy(es)
```

Comment:
The inplace concat will allow us to test concat on a koalas entityset, so I wanted to keep the test as similar to the one above as possible while also allowing it to be used for koalas. 

If you think it'd make more sense, I can rewrite this test to not rely on copying an EntitySet

Reply:
Gotcha - I think if we are going to support concat operation with Dask/Koalas we probably need to make sure we are testing those thoroughly as all of the other tests use only the `pd_es` fixture or xfail with Dask and/or Koalas.

For this particular test I think a comment would be sufficient to note why it is different than the others.

Reply:
I can change the tests to mainly use the inplace concat, that way we can test with koalas. We'll also need to stop relying on the deepcopy calls. 

I'll see if just doing a shallow copy in those cases causes problems. 

For the places where `pd_es` is being used, those are testing the sorting that is performed only on pandas DataFrames, so I can change them to use `es` and check that the dataframes don't actually get sorted.

Reply:
Ended up changing the implementation of deepcopy so that it works for koalas DataFrames as well. Confirmed the deepcopy tests work with this new implementation. It uses `df.ww.copy()`, which I think I like better anyway as a way of retaining the woodwork typing information.